x-pack/plugin/rank-rrf/src/yamlRestTest/resources/rest-api-spec/test/linear/10_linear_retriever.yml

```diff
@@ -1333,3 +1333,605 @@ setup:
+                          boost: 10.0
+                        }
+                      }
+                    }
+                  },
+                  weight: 1.0,
+                  normalizer: minmax
+                }
+              ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "1" }
+  - close_to: { hits.hits.0._score: { value: 12.0, error: 0.001} }
+
+---
+"Linear retriever normalization with zero scores":
+  - requires:
+      cluster_features: [ "linear_retriever.top_level_normalizer" ]
+      reason: "Support for top-level normalizer in linear retriever"
+  - do:
+      search:
+        index: test
+        body:
+          retriever:
+            linear:
+              normalizer: minmax
+              retrievers: [
+                {
+                  retriever: {
+                    standard: {
+                      query: {
+                        constant_score: {
+                          filter: {
+                            term: {
+                              nonexistent_field: "nonexistent_value"
+                            }
+                          },
+                          boost: 5.0
+                        }
+                      }
+                    }
+                  },
+                  weight: 1.0
+                },
+                {
+                  retriever: {
+                    standard: {
+                      query: {
+                        constant_score: {
+                          filter: {
+                            term: {
+                              keyword: {
+                                value: "one"
+                              }
+                            }
+                          },
+                          boost: 10.0
+                        }
+                      }
+                    }
+                  },
+                  weight: 2.0
+                }
+              ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "1" }
+  - close_to: { hits.hits.0._score: { value: 2.0, error: 0.001} }
+
+---
+"Linear retriever error handling - invalid normalizer":
+  - requires:
+      cluster_features: [ "linear_retriever.top_level_normalizer" ]
+      reason: "Support for top-level normalizer in linear retriever"
+  - do:
+      catch: /Unknown normalizer \[invalid\]/
+      search:
+        index: test
+        body:
+          retriever:
+            linear:
+              normalizer: invalid
+              retrievers: [
+                {
+                  retriever: {
+                    standard: {
+                      query: {
+                        match_all: {}
+                      }
+                    }
+                  },
+                  weight: 1.0
+                }
+              ]
+
+---
+"Linear retriever with large score differences and L2 normalization":
+  - requires:
+      cluster_features: [ "linear_retriever.l2_norm", "linear_retriever.top_level_normalizer" ]
```

Comment:
Same comment about multiple cluster features here - if you want to keep the L2 test as an L2 test, it should be a separate test that does not include the top level normalizer.